### PR TITLE
test: add tests for `buildParams`

### LINKED ISSUE #1669: [Test] Adding test for buildParams helper function
With https://github.com/asyncapi/generator/pull/1623 having been merged there are some helper functions which have been created and need to have proper tests for them.

`buildParams` in testing.js is one of them.

### PR DESCRIPTION
**Pull request by harshhgithub**
Resolves #1669 
Description
This PR adds unit tests for the buildParams helper function located in packages/helpers/testing.js.

Changes Introduced
Added test cases to cover different language inputs (including java).
Ensured that the function correctly merges baseParams with config.
Increased coverage for edge cases where clientFileName should/should not be included.

Why is this needed?
Improves test coverage for helper utilities.
Makes future refactoring of buildParams safer by having regression protection.
Helps align with AsyncAPI’s goal of maintaining high-quality, well-tested utilities.

Checklist
 Added new test cases
 All tests pass locally (npm test)
 Coverage increased for buildParams

Screenshot (694)

## Summary by CodeRabbit

- **Tests**
  - Expanded unit test coverage for configuration and language-specific behavior.
  - Added tests for new parameter-building behavior, including conditional inclusion/omission of client file naming (covers lowercase and uppercase language variants).
  - Verified merging of custom parameters with baseParams and base-override behavior.
  - Added handling for missing clientFileName in config.
  - Preserved existing file-discovery tests to guard against regressions.

### CODE REVIEW

File: packages/helpers/test/testing.test.js [hunk 1]
```diff
@@ -30,4 +31,34 @@ describe('listFiles', () => {
...
+    const config = { clientFileName: 'client.js' };
+    const baseParams = { customParam: 'customValue' };
+    const result = buildParams('javascript', config, baseParams);
+    expect(result).toEqual({
+      server: 'echoServer',
+      clientFileName: 'client.js',
+      customParam: 'customValue',
+    });
+  });
+});
```
Comment (Adi-204) [inline packages/helpers/test/testing.test.js]: @harshhgithub you can add 1 more test case in which JAVA is in uppercase so that line https://github.com/asyncapi/generator/blob/master/packages/helpers/src/testing.js#L78 is also tested properly. 

Also add test case where config is an empty object.

Comment (harshhgithub) [reply packages/helpers/test/testing.test.js]: hi @Adi-204 thanks for your review , i have added the testcases as asked

**Review by Adi-204 — APPROVED** [review]
lgtm

**Review by derberg — APPROVED** [review]

**Comment by derberg** [comment]
@harshhgithub thanks and congrats on your first contribution. Don't forget to give the project the star 😉 

since this PR is merged, for others GitHub no longer should require me to accept CI runs for you which should speed up stuff with review and merging

**Comment by harshhgithub** [comment]
> @harshhgithub thanks and congrats on your first contribution. Don't forget to give the project the star 😉
> 
> since this PR is merged, for others GitHub no longer should require me to accept CI runs for you which should speed up stuff with review and merging

thanks @derberg @Adi-204
